src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessFunctions.tsx

```diff
@@ -0,0 +1,135 @@
+// eslint-disable-next-line simple-import-sort/imports
+import React from 'react';
+import styled from '@emotion/styled';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {IntegrationWithConfig, Organization, ServerlessFunction} from 'app/types';
+import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
+import space from 'app/styles/space';
+import Alert from 'app/components/alert';
+import withOrganization from 'app/utils/withOrganization';
+import {t} from 'app/locale';
+import {trackIntegrationEvent} from 'app/utils/integrationUtil';
+
+import IntegrationServerlessRow from './integrationServerlessRow';
+
+type Props = AsyncComponent['props'] & {
+  integration: IntegrationWithConfig;
+  organization: Organization;
+};
+
+type State = AsyncComponent['state'] & {
+  serverlessFunctions: ServerlessFunction[];
+};
+
+class IntegrationServerlessFunctions extends AsyncComponent<Props, State> {
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      serverlessFunctions: [],
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const orgSlug = this.props.organization.slug;
+    return [
+      [
+        'serverlessFunctions',
+        `/organizations/${orgSlug}/integrations/${this.props.integration.id}/serverless-functions/`,
+      ],
+    ];
+  }
+
+  get serverlessFunctions() {
```

Comment:
@scefali I think its better practice to just destructure the data we want from state, 
``` const [severlessFunctions] = this.state;```
Using getters is more verbose.

Reply:
@NisanthanNanthakumar It's more verbose but this abstraction makes it easier to change the shape of your state which is something that might happen here